lib/site.plugin.js

```diff
@@ -1,38 +1,138 @@
 /**
- * Module variables
+ * Module vars
  */
 const root = '/sites';
 
 class SitePlugin {
+
 	/**
 	 * `SitePlugin` constructor.
 	 *
-	 * @param {String} [id] - the plugin ID
+	 * @param {String} [slug] - the plugin slug
 	 * @param {Number|String} sid - site identifier
 	 * @param {WPCOM} wpcom - wpcom instance
 	 * @return {Undefined} undefined
 	 */
-	constructor( id, sid, wpcom ) {
+	constructor( slug, sid, wpcom ) {
 		if ( ! ( this instanceof SitePlugin ) ) {
-			return new SitePlugin( id, sid, wpcom );
+			return new SitePlugin( slug, sid, wpcom );
+		}
+
+		if ( ! slug ) {
+			throw new Error( '`slug` is not correctly defined' );
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
 		this._sid = sid;
-		this.path = `${root}/${this._sid}/plugins`;
 		this.wpcom = wpcom;
+
+		const path = `${root}/${ this._sid }/plugins`;
```

Comment:
Just a suggestion, you don't have to change anything :)

On Wed, Apr 13, 2016 at 8:07 AM, Damián Suárez <notifications@github.com>
wrote:

> In lib/site.plugin.js
> <https://github.com/Automattic/wpcom.js/pull/178#discussion_r59565597>:
>
> >  		this.wpcom = wpcom;
> > +
> > +		const path = `${root}/${ this._sid }/plugins`;
>
> I did it in my way :-o
> I replaced this.path by const path. Let me remove it completely.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/wpcom.js/pull/178/files/7f81138bc016274c905656692cb97d042e0d8e0d#r59565597>
>